### ISSUE TITLE
Release AligNet artifacts (models, datasets) on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [google-deepmind/alignet#4]
Hi @Qwlouse 🤗

Niels here from the open-source team at Hugging Face. I discovered your work through Hugging Face's daily papers as yours got featured: https://huggingface.co/papers/2409.06509.
The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim
the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

It'd be great to make the AligNet model checkpoints and datasets (AligNet Triplet Dataset, Levels Human Similarity Judgment Dataset) available on the 🤗 hub, to improve their discoverability/visibility.
We can add tags so that people find them when filtering https://huggingface.co/models and https://huggingface.co/datasets.

## Uploading models

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverages the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

## Uploading dataset

Would be awesome to make the dataset available on 🤗 , so that people can do:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/your-dataset")
```
See here for a guide: https://huggingface.co/docs/datasets/loading.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the first few rows of the data in the browser.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗

**Comment by Qwlouse** [comment]
Hi @NielsRogge, 
thank you for reaching out. Publishing our models and datasets on HF would be great! 
A couple of questions: 

* **Model upload**: None of our models are in pytorch. They are `flax.linen` models trained with `jax`. I exported the checkpoints via `jax2tf` in the `tensorflow.SavedModel` format. Is there a way to use `jax` or `tf.SavedModels` with your model hub, or would we have to find a matching pytorch implementation and convert the weights?

* **Datasets**: The `alignet` dataset is based on the `imagenet2012` dataset. We could easily upload a dataset that contains the triplets of ids/filenames and corresponding odd-one-out probabilities. But to be usable this would have to be joined with the imagenet dataset (i.e. load the three images corresponding to the ids in the triplet). 
Is there a way to handle that in the data loader, or should we leave that step to the user?

**Comment by NielsRogge** [comment]
Thanks for the quick reply!

* As for model upload, I recommend uploading the raw files: https://huggingface.co/docs/huggingface_hub/en/guides/upload. This can be done either using drag-and-drop or programmatically using the `huggingface_hub` Python SDK.
* The model can then be downloaded (for example within this code repository) using the `hf_hub_download` or `snapshot_download` one-liners. Here's a guide: https://huggingface.co/docs/huggingface_hub/en/guides/download
* as for the dataset, ImageNet-1k is available here: https://huggingface.co/datasets/ILSVRC/imagenet-1k. As can be seen, you can directly see the images and labels in the browser as it's made compatible with the [Dataset viewer](https://huggingface.co/docs/dataset-viewer/en/index). So it'd be cool to do the same for the alignet dataset, however I'm not sure it's worth re-uploading the entire dataset, I'll ping @lhoestq here. 

You definitely have a choice regarding the dataset, either uploading just raw files, or making it compatible with the Datasets library. It is compatible with CSV, Parquet, txt and so on: https://huggingface.co/docs/datasets/loading#local-and-remote-files

**Comment by lhoestq** [comment]
It's generally better to reupload the dataset, otherwise everyone has to write a custom dataloader to use it. Reuploading also makes sure the dataset can be streamed from HF to a dataloader efficiently which is very convenient
